src/ast/variables/LocalVariable.ts

```diff
@@ -33,17 +37,32 @@ export default class LocalVariable extends Variable {
 		super(name);
 		this.declarations = declarator ? [declarator] : [];
 		this.init = init;
+		this.initialisers = [init];
 		this.reassignmentTracker = reassignmentTracker;
 	}
 
-	addDeclaration(identifier: Identifier) {
+	addDeclaration(identifier: Identifier, init: ExpressionEntity) {
 		this.declarations.push(identifier);
+		this.initialisers.push(init);
+	}
+
+	bind() {
+		if (this.bound) return;
+		this.bound = true;
+		if (this.initialisers.length > 1) {
+			this.isReassigned = true;
+			for (const initialiser of this.initialisers) {
+				initialiser.reassignPath(UNKNOWN_PATH);
+			}
+			this.init = UNKNOWN_EXPRESSION;
+		}
 	}
 
 	getLiteralValueAtPath(
 		path: ObjectPath,
 		recursionTracker: ImmutableEntityPathTracker
 	): LiteralValueOrUnknown {
+		if (!this.bound) this.bind();
```

Comment:
I'd prefer to fix root cause if we can... this feels a bit like trying to plug a leak.